From e590822862cddeee64d4397b66b732b56fc74cfa Mon Sep 17 00:00:00 2001
From: Punnaiah Choudary Kalluri <punnaiah.choudary.kalluri@xilinx.com>
Date: Wed, 18 Nov 2015 09:21:17 +0530
Subject: [PATCH 185/188] net: phy: xilinx: Added phy-type property for 1000
 base-x operation

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Added phy-type property for 1000 base-x operation. Based on this phy-type
value driver will not advertise the other speed modes for 1000 base-x
operation.

Signed-off-by: Punnaiah Choudary Kalluri <punnaia@xilinx.com>
(cherry picked from commit 81fd0e4359961fac4d6422776b2b6fb18385388c)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/phy/xilinx_phy.c | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/drivers/net/phy/xilinx_phy.c b/drivers/net/phy/xilinx_phy.c
index 0c817b5..1985949 100644
--- a/drivers/net/phy/xilinx_phy.c
+++ b/drivers/net/phy/xilinx_phy.c
@@ -92,10 +92,30 @@ static int xilinxphy_read_status(struct phy_device *phydev)
 	return 0;
 }
 
+static int xilinxphy_of_init(struct phy_device *phydev)
+{
+	struct device *dev = &phydev->dev;
+	struct device_node *of_node = dev->of_node;
+	u32 phytype;
+
+	if (!IS_ENABLED(CONFIG_OF_MDIO))
+		return 0;
+
+	if (!of_node)
+		return -ENODEV;
+
+	if (!of_property_read_u32(of_node, "xlnx,phy-type", &phytype))
+		if (phytype == XAE_PHY_TYPE_1000BASE_X)
+			phydev->dev_flags |= XAE_PHY_TYPE_1000BASE_X;
+
+	return 0;
+}
+
 static int xilinxphy_config_init(struct phy_device *phydev)
 {
 	int temp;
 
+	xilinxphy_of_init(phydev);
 	temp = phy_read(phydev, MII_BMCR);
 	temp &= XPCSPMA_PHY_CTRL_ISOLATE_DISABLE;
 	phy_write(phydev, MII_BMCR, temp);
-- 
2.0.2

